### DISCUSSION TITLE
Monetize/Subscription Tool for Running Maestro Tests on Own Hardware

### DISCUSSION THREAD

**Discussion by nicholascm** [mobile-dev-inc/Maestro#1187] [Ideas]
Hey everyone,

I think that mobile-dev should find a way to let people run the tests on their own hardware even while still monetizing it. The reason for this, is that it would 

1. enable teams where own hardware is preferred to do that - some companies prefer to have all their machines and their data in Azure, for example.
2. enable us to debug issues on our own and revert back to old versions of Maestro if a new version introduces a breaking issue. Even with the best quality process in place there, I think there will always be some release that cause someone an issue, and downtime for E2E is not good. Allowing teams to "stick with" something that is working for them helps free the mobile-dev team to innovate as much as possible with opportunities to avoid issues for those who don't want to be on the bleeding edge of development. Personally my team cares much more for consistent E2E than latest + greatest features.
3. Perhaps somehow monetizing the runner itself for a flat fee for use on people's own machines would be a nice form of predictable revenue

Anyways - I think this would solve some of the problems my team has had, and I just wanted to throw it out there. 

Thanks for all your work on Maestro !

**Comment by pandoras-toolbox** [comment]
Cloud solutions are a no-go for us because of data privacy. It does not matter what certification your cloud solution has. That has to do with laws, regulations and audit requirements.

If you want to have more customers, you will make it possible to self-host everything.

**Comment by Fishbowler** [comment]
I think we can comfortably say this is now a solved problem :)
